js/packages/ufg-client/test/unit/fetch-resources.spec.ts

```diff
@@ -72,6 +73,39 @@ describe('fetch-resource', () => {
     assert.deepStrictEqual(resource2, mockResource)
   })
 
+  it('fetch with concurrency limitation', async () => {
+    const mockResources = []
+    let count = 0,
+      maxCount = 0
+    for (let i = 0; i < 10; i++) {
+      mockResources.push(
+        makeResource({
+          url: `http://something${i}`,
+          contentType: 'some/content-type',
+          value: Buffer.from('bla'),
+        }),
+      )
+      nock(`http://something${i}`).get('/').reply(200, countServerParallelRequests)
+    }
+
+    const fetchResource = makeFetchResource({fetchConcurrency: 5})
+    const resources = []
+    for (const resource of mockResources) {
+      resources.push(fetchResource({resource}))
+    }
+
+    await Promise.all(resources)
```

Comment:
Use `map` instead of `for` loop
```suggestion
    await Promise.all(mockResources.map(resource => fetchResource({resource})))
```